base/applications/mspaint/history.cpp

```diff
@@ -116,13 +116,24 @@ void ImageModel::ClearHistory()
     m_redoSteps = 0;
 }
 
+void ImageModel::PushImageForUndo()
+{
+    PushImageForUndo(CopyBitmap());
+}
+
 void ImageModel::PushImageForUndo(HBITMAP hbm)
 {
     ATLTRACE("%s: %d\n", __FUNCTION__, m_currInd);
 
+    if (hbm == NULL)
```

Comment:
If I were you I would move that thing into the PushImageForUndo() function instead.
```c++
    HBITMAP hbm = CopyBitmap();
    if (!hbm)
    {  /** Emit the error and return **/ }

    PushImageForUndo(hbm);
}
```
since one could think that someone else in the code could call PushImageForUndo passing a NULL handle to it, for a completely different reason that would not necessarily be due to an out-of-memory condition.

Reply:
Fixed.